fe/fe-core/src/main/java/org/apache/doris/nereids/parser/LogicalPlanBuilder.java

```diff
@@ -576,6 +582,38 @@ public Expression withBetween(PredicateContext ctx, Expression e) {
         return isNotBetween ? new Not(betweenPredicate) : betweenPredicate;
     }
 
+    /**
+     * Generate like predicate.
+     *
+     * @param ctx PredicateContext
+     * @param e Expression
+     * @return Expression
+     */
+    public Expression withLike(PredicateContext ctx, Expression e) {
+        boolean isNotLike = ctx.NOT() != null ? true : false;
+        LikePredicate likePredicate = new LikePredicate(
+                e,
+                expression(ctx.pattern)
+        );
+        return isNotLike ? new Not(likePredicate) : likePredicate;
```

Comment:
in another pr, we refactor this and generate process in withPredicate

Reply:
which pr？

Reply:
done